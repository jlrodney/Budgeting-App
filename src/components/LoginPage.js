import React from 'react';
import { connect } from 'react-redux';
import startLogin from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div>
  <h1 className="header">Budgeting App</h1>
  <h2>This site will allow you to log your expenses and will help you manage your budget</h2>
  <h2>To access your spending please login</h2>
  <button onClick={startLogin}>Login</button>
  </div>
);


export const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
