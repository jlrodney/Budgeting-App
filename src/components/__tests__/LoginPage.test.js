import React from 'react';
import { shallow } from 'enzyme';
import {LoginPage} from '../LoginPage';

describe('LoginPage Component', () => {
  it('should have a button', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper.find('button').text()).toBe("Login")
    expect(wrapper).toMatchSnapshot()
  })

  it('should call login on button click', () => {
    const startLoginSpy = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLoginSpy} />)
    wrapper.find('button').simulate('click');
    expect(startLoginSpy).toHaveBeenCalled();
  })
})
