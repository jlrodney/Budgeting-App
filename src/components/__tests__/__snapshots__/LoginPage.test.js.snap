// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Component should have a button 1`] = `
<div>
  <h1
    className="header"
  >
    Budgeting App
  </h1>
  <h2>
    This site will allow you to log your expenses and will help you manage your budget
  </h2>
  <h2>
    To access your spending please login
  </h2>
  <button>
    Login
  </button>
</div>
`;
